lib/jsonld-signatures.js

```diff
@@ -228,6 +243,14 @@ api.verify = util.callbackify(async function(input, options) {
   if(options.documentLoader) {
     opts.documentLoader = options.documentLoader;
   }
+  // Why expand just to frame again?  The answer is in the proofPurpose
+  // logic.  We can't anticipate all the terms we might operate on for all
+  // available ProofPurpose suites, so we expand once and let them
+  // operate on it in expanded form.
+  // (Each ProofPupose could compact to its own context, though in the case
+  // of ocap-ld, not even which caveats might be used and what the arguments
+  // of them may be is known in advance.)
+  const [expanded] = await jsonld.expand(input, opts);
```

Comment:
A quick comment here ... if we're expanding here and framing/compacting thereafter, pass the expanded input and the "don't expand" flag (or whatever it's called) to the framing/compacting call so we don't pay any extra price.